pallets/asset-index/src/lib.rs

```diff
@@ -1057,7 +1047,7 @@ pub mod pallet {
 			// native asset can't be added
 			Self::ensure_not_native_asset(&asset_id)?;
 			// mint asset into the treasury account
-			T::Currency::deposit(asset_id, &Self::treasury_account(), units)?;
+			T::Currency::deposit(asset_id, &Self::treasury_account(), units)?; // ??? Bryan - where is this token coming from?
```

Comment:
I still don't fully understand. You cannot mint any assets other than PINT token. So there shouldn't be any `deposit` call.

Reply:
Initially, this was intended as governance-approved mint process for PINT and assets alike. 
You're right this doesn't seem to be sound, since there is no guarantee that these assets we're depositing here are also available in the PINT's parachain account on the asset's native chain, right?
You'd recommend that we require that these assets were also send xcm to PINT? so that `deposit` becomes a transfer?

Reply:
Yes I am happy with a transfer from account controlled by pallet or the caller of a tx.